### PR TITLE
Add passive participants and nickname resolution

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add passive participant field to thread storage and proto conversion
- resolve participant nicknames via identity service
- allow thread-filtered unacked message queries
- add coverage for nickname adds and thread filters

## Testing
- buf generate buf.build/agynio/api --path agynio/api/threads/v1 --path agynio/api/notifications/v1 --path agynio/api/identity/v1 --path agynio/api/authorization/v1 --include-imports
- CGO_ENABLED=0 go vet ./...
- CGO_ENABLED=0 go test ./...
- CGO_ENABLED=0 go build ./...

Refs #132

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- add passive participant storage/mapping and identity nickname resolution
- support thread-scoped unacked message queries
- add unit coverage for nickname add/passive mapping and thread filter SQL

## Test & Lint Summary
Commands:
- buf generate buf.build/agynio/api --path agynio/api/threads/v1 --path agynio/api/notifications/v1 --path agynio/api/identity/v1 --path agynio/api/authorization/v1 --include-imports
- CGO_ENABLED=0 go vet ./...
- CGO_ENABLED=0 go test ./...
- CGO_ENABLED=0 go build ./...

Results:
- Tests: passed
- Lint: no errors

**Comment by casey-brooks** [comment]
## Update
- extend CI/devspace buf generate commands to include identity bindings

## Test & Lint Summary
Commands:
- buf generate buf.build/agynio/api --path agynio/api/threads/v1 --path agynio/api/notifications/v1 --path agynio/api/identity/v1 --path agynio/api/authorization/v1 --include-imports
- CGO_ENABLED=0 go vet ./...
- CGO_ENABLED=0 go test ./...
- CGO_ENABLED=0 go build ./...

Results:
- Tests: passed
- Lint: no errors

File: internal/server/server.go [hunk 1]
```diff
@@ -238,6 +264,53 @@ func (s *Server) AckMessages(ctx context.Context, req *threadsv1.AckMessagesRequ
...
+		}
+		if value.ParticipantNickname == "" {
+			return uuid.UUID{}, status.Error(codes.InvalidArgument, "participant.participant_nickname must be provided")
+		}
+		organizationID, err := parseUUID(req.GetOrganizationId())
+		if err != nil {
+			return uuid.UUID{}, status.Errorf(codes.InvalidArgument, "organization_id: %v", err)
+		}
+		if s.identity == nil {
+			return uuid.UUID{}, status.Error(codes.Internal, "identity client is unavailable")
```
Comment (noa-lucent) [inline internal/server/server.go]: [major] Defensive nil check on a required dependency. `identity` is injected via the constructor and is always expected to be non-nil — just like `notifier`, which is used without a nil guard (e.g., line 124). This `if s.identity == nil` branch is a "just-in-case" fallback that masks a misconfiguration bug instead of surfacing it.

Remove this check. If the dependency is missing, the nil-pointer panic at the call site is the correct signal that wiring is broken. This keeps behavior consistent with how other required dependencies are handled.

File: internal/server/server.go [hunk 2]
```diff
@@ -16,12 +18,28 @@ import (
 
 type Server struct {
 	threadsv1.UnimplementedThreadsServiceServer
-	store    *store.Store
+	store    threadStore
 	notifier *notifier.Notifier
```
Comment (noa-lucent) [inline internal/server/server.go]: [minor] `store` and `identity` are now interfaces (good for testability), but `notifier` remains a concrete `*notifier.Notifier`. This inconsistency means the test already passes `nil` for notifier (line 116) and would panic if a code path through `SendMessage` were exercised.

Consider extracting a `notifier` interface as well, to complete the abstraction and enable full unit testing of all server methods.

File: internal/store/messages_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	if len(args) != 2 {
+		t.Fatalf("expected 2 args, got %d", len(args))
+	}
+	if args[0] != participantID {
+		t.Fatalf("expected participant ID arg %s, got %v", participantID, args[0])
+	}
+	if args[1] != 11 {
+		t.Fatalf("expected limit arg 11, got %v", args[1])
+	}
+}
```
Comment (noa-lucent) [inline internal/store/messages_test.go:49]: [minor] Missing test case for the combined scenario: both `threadID` and `cursor` provided simultaneously. This is important because the parameter indexing logic (`paramIndex`) is the most error-prone part of `buildUnackedMessagesQuery` — if the indices are wrong when both filters are active, the query will bind arguments to the wrong placeholders. Add a test that verifies correct `$N` placeholders and argument order when both `threadID != nil` and `cursor != nil`.

File: internal/server/server_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	if !identityStub.called {
+		t.Fatal("expected ResolveNickname to be called")
+	}
+	if resp.GetThread() == nil || len(resp.GetThread().GetParticipants()) != 1 {
+		t.Fatalf("expected 1 participant, got %v", resp.GetThread().GetParticipants())
+	}
+	if !resp.GetThread().GetParticipants()[0].GetPassive() {
+		t.Fatal("expected passive participant to be true")
+	}
+}
```
Comment (noa-lucent) [inline internal/server/server_test.go:155]: [minor] Only the nickname resolution path is tested. Consider adding test cases for:
1. `AddParticipant` with a direct `participant_id` (via the `ParticipantIdentifier_ParticipantId` oneof variant).
2. `AddParticipant` using the deprecated top-level `participant_id` field (the `req.GetParticipant() == nil` branch).
3. Error cases: missing `organization_id` when using nickname, empty nickname, identity resolver returning an error.

These paths contain new branching logic introduced in `resolveParticipantID` and should be covered to prevent regressions.

**Comment by noa-lucent** [comment]
**Note on passive participants and message delivery:**

`loadRecipients` in `internal/store/messages.go` (line 78) selects all non-sender participants without filtering by the new `passive` column. This means passive participants will still appear as message recipients and receive notification events.

If that's intentional (passive participants observe messages but are flagged differently), a brief comment in the code would help future readers. If passive participants should *not* receive messages, the query needs a `AND passive = false` filter.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Overall this is a well-structured PR — clean extraction of the query builder, good use of interfaces for testability, and the migration is straightforward.

**Requesting changes** for one major issue:
- The `s.identity == nil` defensive check in `resolveParticipantID` is a "just-in-case" guard on a required dependency that should be removed to stay consistent with how other dependencies are handled.

**Additional suggestions (non-blocking):**
- Extract a `notifier` interface to complete the testability refactor.
- Add test coverage for the combined threadID + cursor query builder case, and for the other `resolveParticipantID` branches.
- Clarify whether passive participants should receive message notifications.

**Comment by casey-brooks** [comment]
## Update
- remove identity nil guard and add notifier interface for easier mocking
- expand AddParticipant and unacked message query tests
- document passive notification behavior in message send flow

## Test & Lint Summary
Commands:
- buf generate buf.build/agynio/api --path agynio/api/threads/v1 --path agynio/api/notifications/v1 --path agynio/api/identity/v1 --path agynio/api/authorization/v1 --include-imports
- CGO_ENABLED=0 go vet ./...
- CGO_ENABLED=0 go test ./...
- CGO_ENABLED=0 go build ./...

Results:
- Tests: passed
- Lint: no errors

**Review by noa-lucent — APPROVED** [review]
All prior feedback has been addressed:

- **Defensive nil check removed** — `s.identity == nil` guard is gone; consistent with how other dependencies are handled.
- **`notifier` extracted to interface** (`notifierPublisher`) — all three server dependencies are now interface-typed, enabling clean unit testing.
- **Combined threadID + cursor test added** — verifies correct parameter indexing in `buildUnackedMessagesQuery`.
- **Expanded `resolveParticipantID` test coverage** — oneof ID path, legacy field path, and missing org ID error case are all covered.
- **Passive participant clarification** — comment added in `loadRecipients` explaining the design decision.

The stub refactoring to callback-based functions is a nice improvement. Clean update overall. Approved.
